src/utilities/data/Vector.hpp

```diff
@@ -75,21 +76,22 @@ struct UTILITIES_API InterpInfo
   bool extrapolated;  // was point out of range
   unsigned ia, ib;    // indices of two nearest points
   double wa, wb;      // weights of two nearest points
+  double ti;          // length of interval
 };
 
 /** Linear interpolation of the function y = f(x) at point xi. Assumes that x is strictly
  *  increasing. */
-UTILITIES_API InterpInfo interpInfo(const Vector& x, double xi);
+UTILITIES_API InterpInfo interpInfo(const Vector& x, double xi, double ti);
 
 /** Linear interpolation of the function y = f(x) at point xi. Assumes that x is strictly
  *  increasing */
 UTILITIES_API double interp(const Vector& x, const Vector& y, double xi, InterpMethod interpMethod = LinearInterp,
-                            ExtrapMethod extrapMethod = NoneExtrap);
+                            ExtrapMethod extrapMethod = NoneExtrap, double ti = 0.0);
```

Comment:
After scanning the code, I think this is really trying to replace a `boost::optional<double>` only needed for InterpMethod == AverageInterp

Save for actually using a boost::optional<double>, I think you should make the default a negative value, like `-9999.0` (abritrary E+ convention)